app/models/restaurant.rb

```diff
@@ -0,0 +1,18 @@
+class Restaurant < ApplicationRecord
+  include Addressed, Reviewable
+
+  extend FriendlyId
+  friendly_id :name, use: :slugged
+
+  attribute :rating
+
+  has_many :restaurant_genres, dependent: :destroy
+  has_many :genres, through: :restaurant_genres
+
+  validates :name, presence: true
+  validates :max_delivery_time, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  def rating
+    reviews&.average(:rating)&.round || 0
```

Comment:
You should probably save this as an attribute, calculated in `after_save` of a review.

Think of a case where you have 1M reviews and you keep using an average query.

Reply:
@sagish - great catch, thanks.
I am adding this to the after_commit which is a bit tricky since ```Review``` is polymorphic and not sure all ```reviewable``` are implementing the same averaging logic.
I thought of doing this something like:
```ruby
after_commit :update_reviewable_rating

def update_reviewable_rating
  reviewable.update_rating_from_review if reviewable.respond_to?(:update_rating_from_review)
end
```
In general, i'm trying to avoid the ```callbacks``` as they hide logic and in addition in this case not all ```reviewable``` object might implement it, any thoughts on this?